concrete/src/Application/UserInterface/Sitemap/StandardSitemapProvider.php

```diff
@@ -143,22 +142,32 @@ public function getRequestedSiteTree()
             $this->cookieJar->getResponseCookies()->addCookie($cookieKey, $query->get('siteTreeID'));
 
             return $this->siteService->getSiteTreeByID($query->get('siteTreeID'));
-        } elseif ($this->cookieJar->has($cookieKey)) {
-            return $this->siteService->getSiteTreeByID($this->cookieJar->get($cookieKey));
         } else {
-            $site = $this->siteService->getActiveSiteForEditing();
-            $locale = $site->getDefaultLocale();
-            if ($locale && $this->checkPermissions($locale)) {
-                return $locale->getSiteTreeObject();
+            // Check if the site id in $cookieKey is valid
+            $site = null;
+            if ($this->cookieJar->has($cookieKey)) {
+                $site = $this->siteService->getSiteTreeByID($this->cookieJar->get($cookieKey));
             }
+            if (is_object($site)) {
+                return $site;
+            } else {
+                $site = $this->siteService->getActiveSiteForEditing();
```

Comment:
Based on it's docblock it seems this can return `null`. But I guess that's a different problem than what you're trying to fix in this PR.